examples/dmd/wave_equation.cpp

```diff
@@ -0,0 +1,630 @@
+
+void WaveOperator::SetParameters(const Vector &u)
+{
+   delete T;
+   T = NULL; // re-compute T on the next ImplicitSolve
+}
+
+WaveOperator::~WaveOperator()
+{
+   delete T;
+   delete M;
+   delete K;
+   delete c2;
+}
+
+double InitialSolution(const Vector &x)
+{
+   return exp(-x.Norml2()*x.Norml2()*30);
+}
+
+double InitialRate(const Vector &x)
+{
+   return 0.0;
+}
+
+
+int main(int argc, char *argv[])
+{
+   // 1. Parse command-line options.
+   int myid = 0;
+   const char *mesh_file = "../data/star.mesh";
+   const char *ref_dir  = "";
+   int ref_levels = 2;
+   int order = 2;
+   int ode_solver_type = 10;
+   double t_final = 0.5;
+   double dt = 1.0e-2;
+   double speed = 1.0;
+   double ef = 0.9999;
+   int rdim = -1;
+   int windowNumSamples = numeric_limits<int>::max();
+   bool visualization = true;
+   bool visit = false;
+   bool dirichlet = true;
+   int vis_steps = 5;
+   bool use_nonuniform = false;
+
+   int precision = 8;
+   cout.precision(precision);
+
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh",
+                  "Mesh file to use.");
+   args.AddOption(&ref_levels, "-r", "--refine",
+                  "Number of times to refine the mesh uniformly.");
+   args.AddOption(&order, "-o", "--order",
+                  "Order (degree) of the finite elements.");
+   args.AddOption(&ode_solver_type, "-s", "--ode-solver",
+                  "ODE solver: [0--10] - GeneralizedAlpha(0.1 * s),\n\t"
+                  "\t   11 - Average Acceleration, 12 - Linear Acceleration\n"
+                  "\t   13 - CentralDifference, 14 - FoxGoodwin");
+   args.AddOption(&t_final, "-tf", "--t-final",
+                  "Final time; start time is 0.");
+   args.AddOption(&dt, "-dt", "--time-step",
+                  "Time step.");
+   args.AddOption(&speed, "-c", "--speed",
+                  "Wave speed.");
+   args.AddOption(&dirichlet, "-dir", "--dirichlet", "-neu",
+                  "--neumann",
+                  "BC switch.");
+   args.AddOption(&ref_dir, "-r", "--ref",
+                  "Reference directory for checking final solution.");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.AddOption(&visit, "-visit", "--visit-datafiles", "-no-visit",
+                  "--no-visit-datafiles",
+                  "Save data files for VisIt (visit.llnl.gov) visualization.");
+   args.AddOption(&vis_steps, "-vs", "--visualization-steps",
+                  "Visualize every n-th timestep.");
+   args.AddOption(&ef, "-ef", "--energy_fraction",
+                   "Energy fraction for DMD.");
+   args.AddOption(&rdim, "-rdim", "--rdim",
+                   "Reduced dimension for DMD.");
+   args.AddOption(&use_nonuniform, "-nonunif", "--nonunif",       
+                   "-no-nonunif",
+                   "--no-nonunif",
+                   "Use NonuniformDMD.");
+
```

Comment:
Extra line

Reply:
Fixed. NonUniformDMD is not supported yet, so that option should not be present for the time being.